ibis/backends/__init__.py

```diff
@@ -1236,6 +1239,123 @@ def has_operation(cls, operation: type[ops.Value]) -> bool:
+        table_name
+            An optional name to use for the created table. This defaults to
+            a sequentially generated name.
+        **kwargs
+            Additional keyword arguments passed to the pyarrow loading function.
+            See https://arrow.apache.org/docs/python/generated/pyarrow.parquet.read_table.html
+            for more information.
+
+            When reading data from cloud storage (such as Amazon S3 or Google Cloud Storage),
+            credentials can be provided via the `filesystem` argument by creating an appropriate
+            filesystem object (e.g., `pyarrow.fs.S3FileSystem`).
+
+            For URLs with credentials, `fsspec` is used to handle authentication and file access.
+            Pass the credentials using the `credentials` keyword argument. `fsspec` will use these
+            credentials to manage access to the remote files.
+
+        Returns
+        -------
+        ir.Table
+            The just-registered table
+
+        Examples
+        --------
+        Connect to a SQLite database:
+
+        >>> con = ibis.sqlite.connect()
+
+        Read a single parquet file:
+
+        >>> table = con.read_parquet("path/to/file.parquet")
+
+        Read all parquet files in a directory:
+
+        >>> table = con.read_parquet("path/to/parquet_directory/")
+
+        Read parquet files with a glob pattern
+
+        >>> table = con.read_parquet("path/to/parquet_directory/data_*.parquet")
+
+        Read from Amazon S3
+
+        >>> table = con.read_parquet("s3://bucket-name/path/to/file.parquet")
+
+        Read from Google Cloud Storage
+
+        >>> table = con.read_parquet("gs://bucket-name/path/to/file.parquet")
+
+        Read from HTTPS URL
+
+        >>> table = con.read_parquet("https://example.com/data/file.parquet")
+
+        Read with a custom table name
+
+        >>> table = con.read_parquet("s3://bucket/data.parquet", table_name="my_table")
+
+        Read with additional pyarrow options
+
+        >>> table = con.read_parquet("gs://bucket/data.parquet", columns=["col1", "col2"])
+
+        Read from Amazon S3 with secret info
+
+        >>> from pyarrow import fs
+        >>> s3_fs = fs.S3FileSystem(
+        ...     access_key="YOUR_ACCESS_KEY", secret_key="YOUR_SECRET_KEY", region="YOUR_AWS_REGION"
+        ... )
+        >>> table = con.read_parquet("s3://bucket/data.parquet", filesystem=s3_fs)
+
+        Read from HTTPS URL with authentication tokens
+
+        >>> table = con.read_parquet(
+        ...     "https://example.com/data/file.parquet",
+        ...     credentials={"headers": {"Authorization": "Bearer YOUR_TOKEN"}},
+        ... )
+
+        """
+
+        table = self._get_pyarrow_table_from_path(path, **kwargs)
+        table_name = table_name or util.gen_name("read_parquet")
+        self.create_table(table_name, table)
+        return self.table(table_name)
+
+    def _get_pyarrow_table_from_path(self, path: str | Path, **kwargs) -> pa.Table:
```

Comment:
Why can't the implementation of this just be:

```python
return pq.read_table(path)
```

Did you try that already?

Reply:
I tried that in my first commit, it cannot handle all the cases:

such as glob pattern and Parquet files hosted on some uri: i.e HTTPS SFTP 

Pyarrow implements natively the following filesystem subclasses:

[Local FS](https://arrow.apache.org/docs/python/filesystems.html#filesystem-localfs) ([LocalFileSystem](https://arrow.apache.org/docs/python/generated/pyarrow.fs.LocalFileSystem.html#pyarrow.fs.LocalFileSystem))

[S3](https://arrow.apache.org/docs/python/filesystems.html#filesystem-s3) ([S3FileSystem](https://arrow.apache.org/docs/python/generated/pyarrow.fs.S3FileSystem.html#pyarrow.fs.S3FileSystem))

[Google Cloud Storage File System](https://arrow.apache.org/docs/python/filesystems.html#filesystem-gcs) ([GcsFileSystem](https://arrow.apache.org/docs/python/generated/pyarrow.fs.GcsFileSystem.html#pyarrow.fs.GcsFileSystem))

[Hadoop Distributed File System (HDFS)](https://arrow.apache.org/docs/python/filesystems.html#filesystem-hdfs) ([HadoopFileSystem](https://arrow.apache.org/docs/python/generated/pyarrow.fs.HadoopFileSystem.html#pyarrow.fs.HadoopFileSystem))

Reply:
@cpcloud does this make sense to you?